includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -364,13 +367,37 @@ protected function get_row_data( $item ) {
 					$course_title = '<strong><a class="row-title" href="' . esc_url( $url ) . '">' . apply_filters( 'the_title', $item->post_title, $item->ID ) . '</a></strong>';
 				}
 
+				// Fetch learners in course.
+				$course_args           = array(
+					'post_id' => $item->ID,
+					'type'    => 'sensei_course_status',
+					'status'  => 'any',
+				);
+				$course_students_count = Sensei_Utils::sensei_check_for_activity( apply_filters( 'sensei_analysis_course_learners', $course_args, $item ) );
+
+				// Fetch course stats completed lessons and total lessons.
+				$stats                   = Sensei()->course->get_progress_stats( $item->ID );
+				$lesson_completions      = $stats['completed_lessons_count'];
```

Comment:
I don't believe using this function is going to give us an accurate count. We [previously determined](https://github.com/Automattic/sensei/pull/4897#issuecomment-1069325082) that we should consider a lesson to be completed if it has a completion date. It will have a completion date if it has any of the following statuses - `graded`, `ungraded`, `passed`, `failed`, or `complete`.

The `get_progress_stats` function ultimately traces back to [here](https://github.com/Automattic/sensei/blob/master/includes/class-sensei-utils.php#L1602), where you'll notice it's considering an `ungraded` status to be `false` (i.e. not a completed lesson).

It's too risky to change the logic in the `user_completed_lesson` to return `true`, since that function is called in a lot of places and we could break something. We may need to use an alternate solution here.

Perhaps instead, we could do something like [this](https://github.com/Automattic/sensei/blob/a83297fa99d84f4b8a1100c1d3e382f86f0fa362/includes/class-sensei-analysis-overview-list-table.php#L364) to get the completed lesson count.

Reply:
@donnapep @m1r0  do you like my new approach to this calculation: 
[here](https://github.com/Automattic/sensei/pull/4917/files#diff-6ce1dc075c4832d5bf050e621bcfecbab91d97504e8bc565f4f476b1ea2f654dR383)

@donnapep I used post__in, good solution, didn't know it exists :D 
@m1r0 you are right the calculation wasn't working well because I didn't take the statuses in the count.